javascripts/tracking.js

```diff
@@ -130,19 +130,22 @@ piwikUsageTracking.createTrackersIfNeeded = function ()
     }
 
     $(function () {
-        var $rootScope = angular.element(document).injector().get('$rootScope');
-
         $(broadcast).on('locationChangeSuccess', function () {
             trackPageView();
         });
 
-        $rootScope.$on('$locationChangeSuccess', function (event, newUrl, oldUrl) {
+        var watch = window.Vue.watch;
+        var MatomoUrl = window.CoreHome.MatomoUrl;
+        watch(
+          function () { return MatomoUrl.url; },
+          function (newUrl, oldUrl) {
             var urlAnonymizer = createUrlAnonymizer();
 
             if (urlAnonymizer.isPiwik3ReportingUrl() && newUrl !== oldUrl) {
```

Comment:
I'm not very deep into that plugin, but can't we maybe remove that part of the code completely? As the version released from `5.x-dev` branch will be only compatible with Matomo 5, there shouldn't be any checks needed anymore for urls that might look like old Piwik 3 urls. Or is that somehow used to track the admin part maybe? 🤔

Reply:
@sgiehl looks like it's not used, I couldn't see it ever get triggered. I also couldn't see locationChangeSuccess ever get invoked... seems like broadcast.loadAjaxContent isn't really used anymore. Modified it to use the `piwikPageChange` event, and it started tracking pageviews again. Thanks for catching that.

Reply:
As we are trying to get rid of the old name over time and we have the possibility to break bc with Matomo 5, wouldn't it make sense to rename the `piwikPageChange` event to `matomoPageChange` in core?

Reply:
ping @sgiehl ^

Reply:
@diosmosis I've created https://github.com/matomo-org/matomo/pull/19710, feel free to review and merge that one.

Reply:
@sgiehl updated this PR